pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -35,11 +37,12 @@ public DownloadParticipantListRoute() {
     public Object processRequest(Request request, Response response, String userId) throws Exception {
```

Comment:
looking at the processRequest: Can we please try to come up with a way to do counting and processing in one? In a study like TestBoston with 10.000 patients there would need to be 20 calls to ES for counting and another 20 calls for processing.  that seems excessive and it would be a big hit to performance.
Also even if we end up doing it that way, can we please move the whole block into a separate method and not just copy paste the code?

Reply:
Maybe we can introduce Java's parallelism here or Producer/Consumer technique to speed up the process and avoid processing huge amount of data by only one thread